File: proposta-credito-master/src/main/java/com/testecredito/controller/PropostaController.java
package com.testecredito.controller;

import java.net.URI;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.support.ServletUriComponentsBuilder;

import com.testecredito.entity.Proposta;
import com.testecredito.repository.PropostaRepository;


@RestController
@RequestMapping("propostas")
public class PropostaController {

	@Autowired
	private PropostaRepository propostaRepository;	
	
	@Autowired
	private AnaliseCreditoController analiseCreditoController;

	@GetMapping
	public List<Proposta> getAllPropostas() {
		return (List<Proposta>) this.propostaRepository.findAll();
	}

	@GetMapping("/{id}")
	public Proposta getProposta(@PathVariable("id") final Proposta proposta) {
		return proposta;
	}
	
	@GetMapping("/filter/{cpf}")
	public List<Proposta> getByCpf(@PathVariable("cpf") final String cpf) {
		return (List<Proposta>) this.propostaRepository.findByCpf(cpf);
	}

	@PostMapping
	public ResponseEntity<Proposta> addProposta(@RequestBody final Proposta proposta) {
		final Proposta newProposta = this.propostaRepository.save(proposta);		
		

		final URI location = ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}")
				.buildAndExpand(newProposta.getId()).toUri();

		return ResponseEntity.created(location).build();
	}
	
	@PutMapping("/{id}")
	public void analisarProposta(@PathVariable("id") final Long id, @RequestBody final Proposta proposta) {
		this.analiseCreditoController.analisaProposta(id, proposta);
	}

}

File: proposta-credito-master/src/main/resources/static/js/controller/proposta.controller.js
app.controller("propostaController", function($scope, PropostaService){
	$scope.propostas = [];
    $scope.proposta = {};
    $scope.filter = "";
    $scope.saved = false;
    $scope.error = false;
    
    $scope.findAllPropostas = function(){
        console.log("teste list");
		PropostaService.findAllPropostas().then(function(data){            
            $scope.propostas = data;
            console.log($scope.propostas);
		});
    };
    
    $scope.findByCpf = function(){
        console.log("buscou"+ $scope.filter);
        if($scope.filter){
            PropostaService.findByCpf($scope.filter).then(function(data){
			    $scope.propostas = data;
		    });
        }else{
            PropostaService.findAllPropostas().then(function(data){            
                $scope.propostas = data;
                console.log($scope.propostas);
		    });
        }		
	};

    $scope.addProposta = function(){
        console.log("teste add");
		PropostaService.addProposta($scope.proposta).then(function(data){
			$scope.saved = true;
		});		
		$scope.proposta = {};
    };

    $scope.analisarProposta = function(proposta){
        console.log("teste add");
		PropostaService.analisarProposta(proposta).then(function(){
			PropostaService.findByCpf(proposta.cpf).then(function(data){
                $scope.propostas = data;
                $scope.filter = proposta.cpf;
		    });
		});		
		$scope.proposta = {};
    };
    
    $scope.findAllPropostas();

});